src/main/java/study/racing/domain/Name.java

```diff
@@ -1,17 +1,21 @@
 package study.racing.domain;
 
+import study.racing.exception.InvalidCarNameException;
+
 public class Name {
 
+    public static final String INVALID_NAME_MESSAGE = "car name length must not greater than ";
+    public static final int NAME_LENGTH_LIMIT = 5;
+
     private final String name;
 
     public Name(String name) {
+        if (name.length() > NAME_LENGTH_LIMIT) {
+            throw new InvalidCarNameException(INVALID_NAME_MESSAGE + NAME_LENGTH_LIMIT);
+        }
         this.name = name;
     }
```

Comment:
```suggestion
    public Name(String name) {
        validateOrElseThrow(name);
        this.name = name;
    }

    private void validateOrElseThrow(String name) {
        if (name.length() > NAME_LENGTH_LIMIT) {
            throw new InvalidCarNameException(INVALID_NAME_MESSAGE + NAME_LENGTH_LIMIT);
        }
    }
```
중요한 행위는 메서드로 만드는게 좋습니다. 지난 번 코드에서는 메서드를 만들었지만, 생성시점에서 해당 로직을 호출하지 않아서 문제가 있는 인자여도 객체가 생성되기에 문제였던 것이죠 ㅎㅎ 생성하는 시점에서 객체를 생성할 수 있는지 매개변수를 검사하는 로직을 넣어달라는 것을 말하고싶은 것이였습니다. 

이 내용은 이펙티브 자바의 아이템 49에서도 나오는 것인데요 한 번 참고해보세요! ([아이템 49](https://sejoung.github.io/2019/02/2019-02-14-Check_parameters_for_validity/))